eclipse/src/saros/ui/widgets/ConnectionStateComposite.java

```diff
@@ -122,7 +100,8 @@ public void widgetDisposed(DisposeEvent e) {
    * @param state the current connection state or <code>null</code>
    * @param error additional error information or <code>null</code>
    */
-  private void updateLabel(ConnectionState state, Exception error) {
+  private void updateLabel(ConnectionState state, ErrorType error) {
+    if (isDisposed()) return;
```

Comment:
Thanks for violating guidelines.

See https://www.ibm.com/support/knowledgecenter/en/SS8PJ7_9.5.0/org.eclipse.platform.doc.isv/reference/api/org/eclipse/swt/widgets/Widget.html#checkWidget()

Basically this means that you should call this method in every public method of a widget.

It is just bad coding to check if the widget itself has disposed and silently drop any further action (even in private methods).

There is a reason the check is always done in the runable and not in the called method.

Imagine you introduce a public method that internally calls updateLabel and forget the checkWidget call. Nothing will happen if the widget is already disposed and you will silently swallow the wrong access to the widget.

Even if this might not be a big deal the raised exception could be more helpful to figure out what is going wrong instead of watching the UI and pondering why it is not correctly updating.

Reply:
> Thanks for violating guidelines.

I find your tone passively aggressive and would appreciate it if you would change it.

I moved the check from the listeners to this position. And the check was introduced a while ago by yourself:
https://github.com/saros-project/saros/commit/6c6faca6d871d3f7c224961db95aa2e04b0d47b2#diff-66f12eebcb2e00c1fb289219b89fead8R57-R58 So I'm not sure why your argue like that. What is now added is that it is also checked in the constructor execution (which I would say doesn't change something).

---

From your linked doc:
>Currently, it is an error to invoke any method (other than isDisposed()) on a widget that has had its dispose() method called.

What I read is use checkWidget if you would like to have a exception, use isDisposed if you just want to check. Do you see something that should be done in the disposed state?
The check is like before still done in the context of the runnable thread.
Arguing about the visibility of the check in cases of for example additions to the listeners, Ok this is maybe a thing but a reason for this reviews.
Same for errors in new methods and I don't see how it helps in these cases to do the check in the listener. When the widget is disposed the doc says to not call methods on it and there is nothing which is displayed, so what would you update? On dispose code should be handled in a dispose listener.